functions

```diff
@@ -376,6 +376,9 @@ add_firmware() {
             elif [[ -f $fwpath/$fw ]]; then
                 add_file "$fwpath/$fw" "$fwpath/$fw" 644 && r=0
                 break
+            elif [[ -f $fwpath/$fw.zst ]]; then
+                add_file "$fwpath/$fw.zst" "$fwpath/$fw.zst" 644 && r=0
+                break
```

Comment:
I think it would make more logical to order it as xz, zst, uncompressed.

```suggestion
            elif [[ -f $fwpath/$fw.zst ]]; then
                add_file "$fwpath/$fw.zst" "$fwpath/$fw.zst" 644 && r=0
                break
            elif [[ -f $fwpath/$fw ]]; then
                add_file "$fwpath/$fw" "$fwpath/$fw" 644 && r=0
                break
```